### PR TITLE
Feature/39520 - Typescript changes in code

### PR DESCRIPTION
**Pull request by KushChauhanSS**
- Installed dependencies for typescript
- Defined strict types
- Setup tslint, husky, and commitzen 

https://redmine.successive.tech/issues/39520

### CODE REVIEW

File: extraTs/interfaces.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
+interface Permission {
```
Comment (adilsuccessive) [inline extraTs/interfaces.ts]: Each interfaces should start with 'I'
e.g.    Permission can be defined as IPermission

File: extraTs/interfaces.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+    [key: string]: {
+        all: string[];
+        read: string[];
+        write: string[];
+        delete: string[];
+    };
+}
+
+interface UserType {
```
Comment (adilsuccessive) [inline extraTs/interfaces.ts]: define it as IUser, no need to add Type at the end.

**Review by adilsuccessive — CHANGES_REQUESTED** [review]

File: extraTs/interfaces.ts (hunk 1 above)
Comment (KushChauhanSS) [reply extraTs/interfaces.ts]: Done changes

File: extraTs/interfaces.ts (hunk 2 above)
Comment (KushChauhanSS) [reply extraTs/interfaces.ts]: Done changes

**Review by adilsuccessive — APPROVED** [review]

**Pull request merged**